shared/actions/bots.tsx

```diff
@@ -0,0 +1,64 @@
+import * as BotsGen from './bots-gen'
+import * as Saga from '../util/saga'
+import * as Container from '../util/container'
+import * as RPCTypes from '../constants/types/rpc-gen'
+import logger from '../logger'
+import {RPCError} from 'util/errors'
+
+const getFeaturedBots = async (_: Container.TypedState, action: BotsGen.GetFeaturedBotsPayload) => {
+  const {limit, page} = action.payload
+
+  try {
+    const {bots} = await RPCTypes.featuredBotFeaturedBotsRpcPromise({
+      limit: limit ?? 10,
+      offset: (page ?? 0) * (limit ?? 10),
+    })
+    if (!bots || bots.length == 0) {
+      // don't do anything with an empty response from rpc
+      return BotsGen.createSetLoadedAllBots({loaded: true})
+    }
+
+    return [BotsGen.createUpdateFeaturedBots({bots, page}), BotsGen.createSetLoadedAllBots({loaded: false})]
+  } catch (e) {
+    const err: RPCError = e
+    if (Container.isNetworkErr(err.code)) {
+      logger.info('Network error getting featured bots')
+    } else {
+      logger.info(err.message)
+    }
+    return
+  }
+}
+
+const searchFeaturedBots = async (state: Container.TypedState, action: BotsGen.SearchFeaturedBotsPayload) => {
+  const {limit, offset, query} = action.payload
+
+  if (state.chat2.featuredBotsMap.has(query)) {
```

Comment:
search may return more than 1 result even in an exact match on one bot, might not want to cache this this way

Reply:
gotcha, i'll take that out